GitUI/CommandsDialogs/FormEditor.cs

```diff
@@ -116,40 +116,52 @@ private void FormEditor_FormClosing(object sender, FormClosingEventArgs e)
 
         private void toolStripSaveButton_Click(object sender, EventArgs e)
         {
-            try
-            {
-                SaveChanges();
-            }
-            catch (Exception ex)
-            {
-                MessageBox.Show(this, _cannotSaveFile.Text + Environment.NewLine + ex.Message, Strings.Error, MessageBoxButtons.OK, MessageBoxIcon.Error);
-            }
+            SaveChanges();
         }
 
-        private void SaveChanges()
+        /// <summary>
+        /// Saves Changes
+        /// </summary>
+        /// <param name="rethrowException">Allows Exception to be rethrown for custom handling if saving chang fails</param>
+        private void SaveChanges(bool rethrowException = false)
         {
-            if (!string.IsNullOrEmpty(_fileName))
+            try
             {
-                if (fileViewer.FilePreamble is null || Module.FilesEncoding.GetPreamble().SequenceEqual(fileViewer.FilePreamble))
-                {
-                    File.WriteAllText(_fileName, fileViewer.GetText(), Module.FilesEncoding);
-                }
-                else
+                if (!string.IsNullOrEmpty(_fileName))
                 {
-                    using (var bytes = new MemoryStream())
+                    if (fileViewer.FilePreamble is null || Module.FilesEncoding.GetPreamble().SequenceEqual(fileViewer.FilePreamble))
+                    {
+                        using FileStream fs = new FileStream(_fileName, FileMode.Open);
+                        using TextWriter tw = new StreamWriter(fs, Module.FilesEncoding, 1024, true);
+                        tw.Write(fileViewer.GetText());
+                        fs.SetLength(fs.Position);
```

Comment:
This non-obvious workaround (including comments) should be factored out as a common helper function.
Where to put it, @RussKie?

Reply:
I found another location that could benefit from it: ConfigFile's Save(). Other than that, the only other File.WriteAllText that is not in a test is for saving the commit message to a file (I assume these wouldn't normally be hidden).

Reply:
So factor out this function, please.
@RussKie, in which (new) class shall this be moved?